src/main/webapp/user_profile_script.js

```diff
@@ -12,9 +12,145 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+async function getInfo(keyString) {
+    const queryURL = "/tasks/info?key=" + keyString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const info = await response.json();
+    console.log(info[0]);
+    return info[0];
+}
+
+async function deleteTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "OPEN") {
```

Comment:
It might be worth considering not including the delete or edit buttons at all for IN PROGRESS tasks, if they're going to be received with this message. But this works as well!

Reply:
For now I would rather keep those buttons, because for the OPEN tasks, there will be two columns, "EDIT" and "DELETE". It looks strange if the "IN PROGRESS" tasks miss the two columns. I will try to come up with a better way to warn people not pressing the two buttons for the "IN PROGRESS" task in the next commit.

Reply:
Sounds good!